File: Code/C_Code/6_1_Doorbell/Doorbell.c
/**********************************************************************
* Filename    : Doorbell.c
* Description : Make doorbell with buzzer and button.
* Author      : www.freenove.com
* modification: 2024/07/29
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>

#define buzzerPin 12  	//define the buzzerPin
#define buttonPin 21		//define the buttonPin

void main(void)
{
	printf("Program is starting ... \n");
	
	wiringPiSetupGpio();//Initialize wiringPi. Use BCM Number.
	
	pinMode(buzzerPin, OUTPUT); 
	pinMode(buttonPin, INPUT);

	pullUpDnControl(buttonPin, PUD_UP);  //pull up to HIGH level
	while(1){
		
		if(digitalRead(buttonPin) == LOW){ //button is pressed
			digitalWrite(buzzerPin, HIGH);   //Turn on buzzer 
			printf("buzzer turned on >>> \n");
		}
		else {				//button is released 
			digitalWrite(buzzerPin, LOW);   //Turn off buzzer
			printf("buzzer turned off <<< \n");
		}
	}
}

File: Code/Python_GPIOZero_Code/6_1_Doorbell/Doorbell.py
#!/usr/bin/env python3
########################################################################
# Filename    : Doorbell.py
# Description : Make doorbell with buzzer and button
# Author      : www.freenove.com
# modification: 2024/07/29
########################################################################
from gpiozero import Buzzer, Button  
import time

buzzer = Buzzer(12)
button = Button(21)

def onButtonPressed():
    buzzer.on()
    print("Button is pressed, buzzer turned on >>>")
    
def onButtonReleased():
    buzzer.off()
    print("Button is released, buzzer turned on <<<")

def loop():
    button.when_pressed = onButtonPressed
    button.when_released = onButtonReleased
    while True :
        time.sleep(1)
        
def destroy():
    buzzer.close()
    button.close()

if __name__ == '__main__':     # Program entrance
    print ('Program is starting ... ')
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        print("Ending program")
    finally:
        destroy()